src/tools/compiletest/src/runtest.rs

```diff
@@ -1520,8 +1520,10 @@ impl<'test> TestCx<'test> {
         };
         rustc.arg(input_file);
 
-        // Use a single thread for efficiency and a deterministic error message order
-        rustc.arg("-Zthreads=1");
+        if !self.props.parallel_front_end {
+            // Use a single thread for efficiency and a deterministic error message order
+            rustc.arg("-Zthreads=1");
```

Comment:
Is this directive necessary? Is this overriding the compile flags from existing tests that use `//@ compile-flags: -Zthreads=45`?